packages/serverless-orchestration/src/ServerlessSpoke.js

```diff
@@ -27,9 +27,10 @@ const waitForLoggerDelay = process.env.WAIT_FOR_LOGGER_DELAY || 5;
 const maxBuffer = 8388608;
 
 spoke.post("/", async (req, res) => {
-  // Use a custom logger if provided. Otherwise, initialize a local logger.
+  // Use a custom logger if provided. Otherwise, initialize a local logger with a run identifier if passed from the Hub.
   // Note: no reason to put this into the try-catch since a logger is required to throw the error.
-  const logger = customLogger || createNewLogger();
+  const logger =
+    customLogger || createNewLogger(undefined, undefined, undefined, req.body?.environmentVariables?.RUN_IDENTIFIER);
```

Comment:
what do all the undefineds mean?

Reply:
The undefined values are just there to use the default args for these parameters.

Before, we used defaults for all params. Now we want to specify one of the later ones, so we have to provide default args for the ones that come before it.

Do you suggest an alternative?

Reply:
no was curious